### ISSUE TITLE
Typeorm for mysql in Browser

### ISSUE THREAD

**Issue by aammfe** [typeorm/typeorm#2874]
**Issue type:**

[ X] question
[ ] bug report
[ ] feature request
[ ] documentation issue

**Database system/driver:**

[ ] `cordova`
[ ] `mongodb`
[ ] `mssql`
[X ] `mysql` / `mariadb`
[ ] `oracle`
[ ] `postgres`
[ ] `sqlite`
[ ] `sqljs`
[ ] `react-native`
[ ] `expo`

**TypeORM version:**

[ ] `latest`
[ ] `@next`
[ ] `0.x.x` (or put your version here)

Hello!!
I m confused right now.
we can not use typeorm for mysql in browser 0k
can we use it in electron and what dbs we can access ? its also a browser but there we have access to file system etc.

basically I want to create application in electron using angular.
and I want to access database directly from it.

we can also use [http](https://mysqlserverteam.com/mysql-5-7-labs-inserting-updating-and-deleting-records-via-http/) to query mysql.

in theory it should work.

please Help !

**Comment by Kononnable** [comment]
If I remember correctly electron has node running under it, so all supported by typeorm drivers can be used with it.

**Comment by aammfe** [comment]
Thanx for quick replay , 

I'm having issues with angular and typeorm together, it would be great to have an example.

**Comment by elthrasher** [comment]
I haven't tried it, but there are many examples of using electron and mysql together. For example: https://github.com/techiediaries/electron-mysql-demo

Using typeorm won't be much different. What issues are you having?

**Comment by aammfe** [comment]
issue is **ng eject** is disabled in angular cli. means I could not customize angular project.
I m using typeorm and angular together and having following errors.

WARNING in ./node_modules/typeorm/platform/PlatformTools.js 131:23-85
Critical dependency: the request of a dependency is an expression

WARNING in ./node_modules/typeorm/platform/PlatformTools.js 126:27-40
Critical dependency: the request of a dependency is an expression

WARNING in ./node_modules/typeorm/platform/PlatformTools.js
Module not found: Error: Can't resolve 'mongodb' in 'E:\Angular workspace\typeorm\with-webpack-first\node_modules\typeorm\platform'

WARNING in ./node_modules/typeorm/platform/PlatformTools.js
Module not found: Error: Can't resolve 'mssql' in 'E:\Angular workspace\typeorm\with-webpack-first\node_modules\typeorm\platform'

WARNING in ./node_modules/typeorm/platform/PlatformTools.js
Module not found: Error: Can't resolve 'mysql' in 'E:\Angular workspace\typeorm\with-webpack-first\node_modules\typeorm\platform'

WARNING in ./node_modules/typeorm/platform/PlatformTools.js
Module not found: Error: Can't resolve 'mysql2' in 'E:\Angular workspace\typeorm\with-webpack-first\node_modules\typeorm\platform'

WARNING in ./node_modules/typeorm/platform/PlatformTools.js
Module not found: Error: Can't resolve 'oracledb' in 'E:\Angular workspace\typeorm\with-webpack-first\node_modules\typeorm\platform'

WARNING in ./node_modules/typeorm/platform/PlatformTools.js
Module not found: Error: Can't resolve 'pg' in 'E:\Angular workspace\typeorm\with-webpack-first\node_modules\typeorm\platform'

WARNING in ./node_modules/typeorm/platform/PlatformTools.js
Module not found: Error: Can't resolve 'pg-native' in 'E:\Angular workspace\typeorm\with-webpack-first\node_modules\typeorm\platform'

WARNING in ./node_modules/typeorm/platform/PlatformTools.js
Module not found: Error: Can't resolve 'pg-query-stream' in 'E:\Angular workspace\typeorm\with-webpack-first\node_modules\typeorm\platform'

WARNING in ./node_modules/typeorm/driver/react-native/ReactNativeDriver.js
Module not found: Error: Can't resolve 'react-native-sqlite-storage' in 'E:\Angular workspace\typeorm\with-webpack-first\node_modules\typeorm\driver\react-native'

WARNING in ./node_modules/typeorm/platform/PlatformTools.js
Module not found: Error: Can't resolve 'redis' in 'E:\Angular workspace\typeorm\with-webpack-first\node_modules\typeorm\platform'

WARNING in ./node_modules/typeorm/platform/PlatformTools.js
Module not found: Error: Can't resolve 'sqlite3' in 'E:\Angular workspace\typeorm\with-webpack-first\node_modules\typeorm\platform'

ERROR in ./node_modules/mkdirp/index.js
Module not found: Error: Can't resolve 'fs' in 'E:\Angular workspace\typeorm\with-webpack-first\node_modules\mkdirp'
ERROR in ./node_modules/typeorm/platform/PlatformTools.js
Module not found: Error: Can't resolve 'fs' in 'E:\Angular workspace\typeorm\with-webpack-first\node_modules\typeorm\platform'
ERROR in ./node_modules/mkdirp/index.js
Module not found: Error: Can't resolve 'path' in 'E:\Angular workspace\typeorm\with-webpack-first\node_modules\mkdirp'
ERROR in ./node_modules/typeorm/platform/PlatformTools.js
Module not found: Error: Can't resolve 'path' in 'E:\Angular workspace\typeorm\with-webpack-first\node_modules\typeorm\platform'
ERROR in ./node_modules/cli-highlight/node_modules/parse5/lib/parser/parser_stream.js
Module not found: Error: Can't resolve 'stream' in 'E:\Angular workspace\typeorm\with-webpack-first\node_modules\cli-highlight\node_modules\parse5\lib\parser'
ERROR in ./node_modules/cli-highlight/node_modules/parse5/lib/sax/index.js
Module not found: Error: Can't resolve 'stream' in 'E:\Angular workspace\typeorm\with-webpack-first\node_modules\cli-highlight\node_modules\parse5\lib\sax'
ERROR in ./node_modules/cli-highlight/node_modules/parse5/lib/sax/dev_null_stream.js
Module not found: Error: Can't resolve 'stream' in 'E:\Angular workspace\typeorm\with-webpack-first\node_modules\cli-highlight\node_modules\parse5\lib\sax'
ERROR in ./node_modules/cli-highlight/node_modules/parse5/lib/serializer/serializer_stream.js
Module not found: Error: Can't resolve 'stream' in 'E:\Angular workspace\typeorm\with-webpack-first\node_modules\cli-highlight\node_modules\parse5\lib\serializer'
ERROR in ./node_modules/typeorm/platform/PlatformTools.js
Module not found: Error: Can't resolve 'stream' in 'E:\Angular workspace\typeorm\with-webpack-first\node_modules\typeorm\platform'

![image](https://user-images.githubusercontent.com/11896817/46350615-9c8d2c80-c66e-11e8-9346-f912c86144f5.png)

**Comment by pleerock** [comment]
webpack shouldn't pickup typeorm files for your browser (frontend) app, so I guess as it usually happens with shared frontend/backend you have mixed some files or imports
